[Docs] Add user level documentation for Expressions (#11990)
KratosMultiphysics/Kratos: docs/pages/Kratos/Expressions/IOs/Nodal_Position_Expression_IO.md

---
title: Nodal Position Expression IO
keywords: 
tags: [nodal position, initial, current, expressions]
sidebar: kratos_expressions
summary: 
---

## Introduction

> ```NodalPositionExpressionIO``` allows writing and reading nodal positions of a given model part. This can be used on following configurations:

Reviewer: ```suggestion
```NodalPositionExpressionIO``` allows writing and reading nodal positions of a given model part. This can be used on following the configurations:
```

* Initial -> This uses the initial nodal position to read and write.

> * Current -> This uses the current position of the nodes to read and write.

Reviewer: ```suggestion
* Initial -> initial positions of the node
* Current -> positions of the nodes in the current configuration
```

Kratos.Expression.NodalPositionExpressionIO.Read(current_nodal_expression,Kratos.Configuration.Current)

print("initial:\n", initial_nodal_expression.Evaluate())
print("current:\n", current_nodal_expression.Evaluate())

# now do some arithmatics with the exps
initial_nodal_expression *= 2
current_nodal_expression *= 2

# now write the modified nodal positions back to nodes
Kratos.Expression.NodalPositionExpressionIO.Write(initial_nodal_expression,Kratos.Configuration.Initial)
Kratos.Expression.NodalPositionExpressionIO.Write(current_nodal_expression,Kratos.Configuration.Current)

# now read nodal positions
for node in model_part.Nodes:
    print(f"node_id: {node.Id}, initial: [{node.X0}, {node.Y0}, {node.Z0}], current: [{node.X}, {node.Y}, {node.Z}]")
```

## Using expressions without the model parts
The ```NodalExpression```, ```ConditionExpression``` and ```ElementExpression``` has an expression which can be directly used if required. The advantage of working
with the ```Expression``` directely is, then it is not bound to a model part of a ```DataValueContainer```. Hence, these expressions can be interchanged if required in

> advanced use cases. Following code snippet shows how to use bare ```Expressions```.

Reviewer: same
